jsonknife.c

```diff
@@ -745,28 +760,38 @@ void reduce_timestamptz(void *acc, JsonbValue *val){
 
 	if(val != NULL && val->type == jbvString) {
 
-		Datum date = date_bound(val->val.string.val, val->val.string.len, dacc->minmax);
+		Datum* datum = date_bound(val->val.string.val, val->val.string.len, dacc->minmax);
+		if (datum == NULL) {
+			return;
+		}
+
+		/* elog(INFO, "reduce_timestamptz %s %ld %ld", val->val.string.val, *datum); */
 
 		if(dacc->minmax == min) {
-			if(dacc->acc != 0){
-				int gt = DirectFunctionCall2(timestamptz_cmp_timestamp, date, dacc->acc);
+			if(dacc->acc != NULL){
+				int gt = DirectFunctionCall2(timestamptz_cmp_timestamp, *datum, *dacc->acc);
 				/* elog(INFO, "compare %d", gt); */
 				if(gt < 0) {
-					dacc->acc = date;
+					pfree(dacc->acc);
```

Comment:
pfree is not neccessary. Postgres deallocs any memory allocated with palloc when transaction ends. (2)